packages/swi-prolog/build.sh

```diff
@@ -0,0 +1,85 @@
+                           ########################################
+                           #          Set variables               #
+                           ########################################
+_TMP_DIR=${TERMUX_PREFIX}/../../cache
+TERMUX_PKG_HOMEPAGE=https://swi-prolog.org/
+TERMUX_PKG_DESCRIPTION="Most popular and complete prolog implementation"
+TERMUX_PKG_VERSION=7.7.23
+TERMUX_PKG_REVISION=3
+TERMUX_PKG_SHA256="79531c2604fbf32c32b35df4784ea4cf4a0f675ab68b041bf4ff6e50bcd38309"
+TERMUX_PKG_SRCURL=http://www.swi-prolog.org/download/devel/src/swipl-${TERMUX_PKG_VERSION}.tar.gz
+
+TERMUX_PKG_DEPENDS="readline, libgmp, libcrypt, pcre, libarchive, libyaml, libjpeg-turbo, ncurses, ncurses-ui-libs, ossp-uuid"
+
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-DINSTALL_DOCUMENTATION=OFF
+-DUSE_GMP=ON
+-DSWIPL_NATIVE_FRIEND=${TERMUX_PKG_HOSTBUILD_DIR}
+-DUNIX_SHELL=${TERMUX_PREFIX}/bin/sh
+-DSWIPL_TMP_DIR=${_TMP_DIR}
+-DSWIPL_INSTALL_IN_LIB=ON
+-DSWIPL_PACKAGES_BDB=OFF
+-DSWIPL_PACKAGES_ODBC=OFF
+-DSWIPL_PACKAGES_QT=OFF
+-DSWIPL_PACKAGES_X=OFF
+-DINSTALL_TESTS=ON
+-DSYSTEM_CACERT_FILENAME=${TERMUX_PREFIX}/etc/tls/cert.pem"
+
+TERMUX_PKG_FORCE_CMAKE=true
+TERMUX_PKG_HOSTBUILD=true
+
+                           ########################################
+                           #          Build in Host               #
+                           #   (in addition to crosscompiling)    #
+                           ########################################
+# We do this to produce:
+# a native host build to produce
+# boot<nn>.prc, INDEX.pl, ssl cetificate tests,
+# SWIPL_NATIVE_FRIEND tells SWI-Prolog to use 
+# this build for the artifacts needed to build the
+# Android version
+termux_step_host_build () {
+        termux_setup_ninja
+        termux_setup_cmake
+
+        sudo dpkg --add-architecture i386
```

Comment:
~I think i686 is enough. I'll fix this.~
**Update**: i686 doesn't work because the zlib1g-dev:i686 pkg does not exist in ubuntu.

What do you think about doing this in two stages to keep things clean:

1. Keep the `dpkg --add-architecture` in the SWI-Prolog PR for now, until we get the SWI-Prolog merged.
2. On a separate PR provide a patch for `setup-ubuntu.sh` once SWI-Prolog is merged. 

The reason is that I think `setup-ubuntu.sh` has to do with all packages  (not just swi-prolog), so I would rather keep a clear separation of concerns to guarantee better stability for termux and easier maintenance. Let me know what you think.

Reply:
> because the zlib1g-dev:i686 pkg does not exist in ubuntu.

Oops, my mistake. 

Host-build fails for me on arm with 
```
/usr/include/openssl/hmac.h:13:11: fatal error: openssl/opensslconf.h: No such file or directory
```
So I guess libssl-dev:i386 is needed as well

Reply:
Fixing that gives 
```
FAILED: home/boot32.prc 
cd /home/builder/.termux-build/swi-prolog/build/src && /home/builder/.termux-build/_cache/cmake-3.13.1/bin/cmake -E remove -f /home/builder/.termux-build/swi-prolog/build/home/boot32.prc && /home/builder/.termux-build/swi-prolog/build/src/swipl -q -O -o /home/builder/.termux-build/swi-prolog/build/home/boot32.prc -b /home/builder/.termux-build/swi-prolog/build/home/boot/init.pl
/bin/sh: 1: /home/builder/.termux-build/swi-prolog/build/src/swipl: Exec format error
```
The host tries to run a binary built for arm (weird that aarch64 works though, should be the same problem there(?)).

Reply:
It acually also needs openssl installed on the host (it needs to run the openssl binary to produce some test files). But if it doesn't find it, it will not build the test files.

The error you are getting about `exec format error` happens when you have an old `host-build` directory with the  wrong word size (e.g. the host-build directory was 64bit when building for arm, which is 32bit). So the swi-prolog tries to use the current build (which won't run on the host, because it is an `arm` binary. This is solved by removing the `host-build` directory.

For this reason we remove the `host-build` directory after the install, but since the build had failed for you, the `host-build` directory was not removed automatically.

I'll push a patch with openssl and libssl-dev:i386